packages/nimble-components/src/table/specs/cell-state-when-scrolling-hld.md

```diff
@@ -0,0 +1,111 @@
+# Nimble Table: Handling Cell State when Scrolling
+
+## Problem Statement
+
+Since the Nimble Table uses virtualization, if the table contains more rows than can be shown in the current table viewport, we will re-use the DOM elements / controls within the table cells as the user scrolls through the table. This is handled by the TanStack virtualizer determining what row indexes are visible at the current scroll position, and then we re-bind the table cell views to the new cell state (including the row data). We need to ensure that cell state isn't incorrectly preserved/ applied to the cells once they represent different row values.
+
+Some of the cell state that would incorrectly apply to new rows (once the user scrolls) is:
+
+-   Selected text (which can span multiple rows), for both editable and non-editable cells
+-   Text cursor position (for editable controls in table cells)
+-   Pending input (for editable controls in table cells)
+-   Open popups (action menu or select/combobox cells whose popup is open)
+
+## Links To Relevant Work Items and Reference Material
+
+[#997: Table virtualization: Clear/Commit (and potentially save/restore) row/cell state when scrolling](https://github.com/ni/nimble/issues/997)
+
+## Implementation / Design
+
+When the user scrolls the table, we will:
+
+-   'Blur' (Lose focus) from any active/focused control in a table cell. Generally this means that whatever text the user began to change in that control will be committed, not discarded.
```

Comment:
In playing with your prototype, if I have a pending edit in one of the text fields and then start scrolling, the pending edit is discarded. That appears to contradict what's written here, but is that just a limitation of the prototype? I'm guessing we don't have any code that handles the blur and updates the data source?

Reply:
Correct, there's just no code in the prototype that updates the data source based on the edits.

Reply:
Maybe we can phrase this one as the table will notify a column plugin that that they will lose focus. The recommended behavior for plugins is to commit changes when appropriate and blur.

Reply:
Updated wording in the HLD. The updated prototype example also no longer uses an editable text box, so the point no longer applies.